pyam/_ops.py

```diff
@@ -70,14 +111,33 @@ def _op_data(df, name, method, axis, fillna=None, args=(), **kwds):
                 kwds[key] = _data[key]
 
     # apply method and check that returned object is valid
-    _value = method(*_args, **kwds)
-    if not isinstance(_value, pd.Series):
+    result = method(*_args, **kwds)
+    if not isinstance(result, pd.Series):
         msg = f"Value returned by `{method.__name__}` cannot be cast to an IamDataFrame"
-        raise ValueError(f"{msg}: {_value}")
-
-    # insert the index level and reset the series name
+        raise ValueError(f"{msg}: {result}")
+
+    # separate pint quantities into numerical value and unit (as index)
+    if ignore_units is False:
+        rename_args = ("dimensionless", "")
+        _value = pd.DataFrame(
+            [[i.magnitude, str(i.units).replace(*rename_args)] for i in result.values],
```

Comment:
Pint can return the unit as `EJ / yr` instead of `exajoule / year` with this code snipplet
```python
# format the the unit as string 'EJ / yr' 
unit = '{:~}'.format(c.units)
```
To implement it, the line should read something like
```python
[[i.magnitude, '{:~}'.format(str(i.units)).replace(*rename_args)] for i in result.values],
```
I'm not sure, if the `str()` is still neded then and how it returns dimensionless units

Reply:
Implemented.